py/fiberassign/fba_tertiary_io.py

```diff
@@ -0,0 +1,631 @@
+
+    Returns:
+        prio:  Table() structured array
+    """
+    if not os.path.isfile(priofn):
+        msg = "missing {}; exiting".format(priofn)
+        log.error(msg)
+        raise IOError(msg)
+    prio = Table.read(priofn)
+    return prio
+
+
+def assert_tertiary_targ(prognum, targfn):
+    """
+    Verifies that a tertiary PROGNUM input targets file is correctly formatted.
+
+    Args:
+        prognum: the tertiary PROGNUM (int)
+        targfn: full path to the tertiary PROGNUM input targets file (string)
+
+    Notes:
+        Will raise an error if the file is not correctly formatted.
+    """
+    # AR read
+    targ, targhdr = read_targfn(targfn)
+
+    # AR check all required keys are there
+    miss_keys = []
+    for key in req_keys:
+        if key not in targ.dtype.names:
+            miss_keys.append(key)
+    if len(miss_keys) > 0:
+        msg = "{} keys are missing from {}; exiting".format(",".join(miss_keys), targfn)
+        log.error(msg)
+        raise IOError(msg)
+
+    # AR check all required header keywords are there
+    miss_hdrkeys = []
+    for hdrkey in req_hdrkeys:
+        if hdrkey not in targhdr:
+            miss_hdrkeys.append(hdrkey)
+    if len(miss_hdrkeys) > 0:
+        msg = "{} keys are missing from the header of {}; exiting".format(
+            ",".join(miss_hdrkeys), targfn
+        )
+        log.error(msg)
+        raise IOError(msg)
+
+    # AR check that the required header keywords are as allowed
+    for hdrkey in allow_hdrkeys:
+        if targhdr[hdrkey] not in allow_hdrkeys[hdrkey]:
+            msg = "hdr['{}']={} not in allowed values ({})".format(
+                hdrkey,
+                targhdr[hdrkey],
+                allow_hdrkeys[hdrkey],
+            )
+            log.error(msg)
+            raise IOError(msg)
+
+    # AR check TARGETID
+    targetids = encode_targetid(
+        release=8888, brickid=prognum, objid=np.arange(len(targ))
+    )
+    sel = targ["TARGETID"] != targetids
+    if sel.sum() > 0:
+        msg = "TARGETID is not as expected, i.e. encode_targetid(release=8888, brickid={}, objid=np.arange(len(targ)))".format(
+            prognum,
+        )
+        log.error(msg)
+        raise IOError(msg)
+
+    # AR check RA, DEC
+    sel = (targ["RA"] >= 0) & (targ["RA"] < 360)
+    sel &= (targ["DEC"] >= -90) & (targ["DEC"] <= 90)
+    if sel.sum() != len(targ):
+        msg = "{} targets do not verify 0 <= RA < 360 and -90 <= DEC <= 90".format(
+            len(targ) - sel.sum()
+        )
+        log.error(msg)
+        raise IOError(msg)
+
+    # AR check PMRA, PMDEC, REF_EPOCH
+    sel = (np.isfinite(targ["PMRA"])) & (np.isfinite(targ["PMDEC"]))
+    if sel.sum() != len(targ):
+        msg = "{} targets do have not finite PMRA or PMDEC".format(
+            len(targ) - sel.sum()
+        )
+        log.error(msg)
+        raise IOError(msg)
+    sel = (np.isfinite(targ["REF_EPOCH"])) & (targ["REF_EPOCH"] > 0)
```

Comment:
I ve required `REF_EPOCH > 1900`.

I m not super-familiar with the existing habits with `REF_EPOCH`, though, with checking the secondary files, I see that some have lots of different values:
```
targets-dark-secondary.fits [1998.723  1998.873  1998.876  ... 2019.7123 2019.7316 2019.7479]
```
I could imagine that the provider would prefer to keep the original coordinates and `REF_EPOCH` from their catalog, rather than move those to some (Gaia) values.

in the case of a grossly wrong `REF_EPOCH`, I guess PlateMaker will complain, but I don t remember if there are other consequences than just not using that fiber.

Reply:
Sounds good. I'm surprised there are so many choices for `REF_EPOCH`, so thanks for checking that.